clang/include/clang/Basic/Cuda.h

```diff
@@ -50,6 +50,10 @@ const char *CudaVersionToString(CudaVersion V);
 // Input is "Major.Minor"
 CudaVersion CudaStringToVersion(const llvm::Twine &S);
 
+// We have a name conflict with sys/mac.h on AIX
+#ifdef SM_32
+#undef SM_32
+#endif
```

Comment:
Ugh. What could possibly go wrong, if someone who needed the original definition of SM_32 ends up transitively including this header and losing the macro definition?

A beeter way to handle it as a workaround would be to push the macro definition, undef it, and then pop it back at the end of the header.

Even better would be to add prefixes to the macros and/or the enum here to disambiguate them

Reply:
We already do that, the problem is that `CudaArch::SM_32` will still invoke the preprocessorr, see https://godbolt.org/z/84xKej5K9. We can't do this from a header level, we'd need to do it around every single use as far as I know, which might be doable?

Reply:
We could always just make all of these lower case instead?

Reply:
> We could always just make all of these lower case instead?

That would be odd. LLVM style wants them to be CamelCased.
This enum is rarely used, so renaming them to something more CUDA/NVPTXspecific would be best, IMO.
E.g `NVSM_32` 

Or we could rename only `SM_32`. The constant is rather inconsequential and is used in a few places only. Renaming it to `_SM_32` with a comment that AIX headers have `SM_32` defined.

Reply:
I mean, modern CUDA installations don't even allow you to build for anything older than `sm_52` now. We could also just delete those enums entirely.

Reply:
Deprecating and removing support for old GPUs needs to be done, but it's not going to happen here and now, so we still need a better short-term fix. 

Undefining a macro set by external headers is not it.

Reply:
Just naming it like `_SM_32` works for me, with a note that they'll get removed in the future.

Reply:
`_SM_32` is a reserved identifier so we should not use that as a name.

Reply:
Good point, totally forgot that `_<uppercase>` was reserved (Would've been nice if the AIX headers followed that rule).

Reply:
SGTM. Thank you for taking care of this issue.

On a side note, do we know if there's a way to file a bug for AIX? They should not be setting macros with names that could conceivably be defined by a user. In theory. I think normally they should be double-underscore-prefixed.

Reply:
Done in https://github.com/llvm/llvm-project/pull/88779